packages/design-system/src/components/web-components/preactement/define.ts

```diff
@@ -138,6 +145,40 @@ function setupElement<T>(component: ComponentFunction<T>, options: IOptions = {}
   };
 }
 
+function proxyEvents(props, eventNames, CustomElement) {
```

Comment:
I think this function could use some comments now that it's working. It could use a high-level explanation of what it sets out to do. We should probably give attribution to the inspiration for this function too.